File: LeetCodeKotlin/InvertBinaryTree_226.kt
class InvertBinaryTree_226 {
    fun invertTree(root: TreeNode?): TreeNode? {
        if (root == null) return null

        // Swap left and right subtrees
        val temp = root.left
        root.left = invertTree(root.right)
        root.right = invertTree(temp)

        return root
    }
}

File: LeetCodeCpp/problems/trees/InvertBinaryTree_226.cpp
#include "InvertBinaryTree_226.h"

#include <utility>

using namespace std;

TreeNode<int>* InvertBinaryTree_226::invertTree(TreeNode<int>* root) {
    if (root == nullptr) {
        return nullptr;
    }

    swap(root->left, root->right);

    invertTree(root->left);
    invertTree(root->right);

    return root;
}
